.github/workflows/master_push.yml

```diff
@@ -17,18 +31,47 @@ jobs:
   tag-build-push:
     runs-on: ubuntu-latest
     steps:
+      - name: Set part of semantic version to bump
+        id: controls
+        run: |
+          SEMVER_PART=""
+          CHECKOUT_BRANCH="$GITHUB_REF"
+          if ${{github.event_name == 'push' }}; then
+            SEMVER_PART="patch"
+          elif ${{github.event_name == 'workflow_dispatch' }}; then
+            SEMVER_PART=${{ github.event.inputs.bump }}
+            CHECKOUT_BRANCH=${{ github.event.inputs.branch }}
+          fi
+          echo ::set-output name=semver-part::$SEMVER_PART
+          echo ::set-output name=checkout-branch::$CHECKOUT_BRANCH
       - name: Checkout current code
         uses: actions/checkout@master
-      - name: Bump version and push tag
+        with:
+          ref: ${{ steps.controls.outputs.checkout-branch }}
+      - name: Skip version bump merges
+        id: skiptest
+        uses: ./.github/actions/bump-skip
+        with:
+          event-name: ${{ github.event_name }}
+      - name: Bump the tag to a new version
+        if: steps.skiptest.outputs.is-bump == 'no'
```

Comment:
I wish there were a way around all these if's. Maybe we could split the job into two and only call the second part if it's not a bump. I.e. do the bifurcation on the input filter of the job instead of at all the steps.

Reply:
If you figure out a way that works, let me know. The continue to add features to GHA, so maybe we can do that someday.